lib/Controller/WopiController.php

```diff
@@ -485,10 +485,21 @@ public function putFile($fileId,
 
 			$content = fopen('php://input', 'rb');
 
+			// Get expected FileSize after writing
+			$storage = \OC_Helper::getStorageInfo($path);
```

Comment:
Unless I oversee something, it should be possible to use the file parent to get the free space through the public (OCP) API:

Something like:
```
$freeSpace = $file->getParent()->getFreeSpace();
```